sql/pipelines/src/test/scala/org/apache/spark/sql/pipelines/graph/SqlPipelineSuite.scala

```diff
@@ -0,0 +1,788 @@
+    val sqlGraphRegistrationContext = new SqlGraphRegistrationContext(graphRegistrationContext)
+
+    val ex = intercept[ParseException] {
+      sqlGraphRegistrationContext.processSqlFile(
+        sqlText = """CREATE STREAMING TABLE a
+                    |LOCATION "/path/to/table"
+                    |AS SELECT * FROM range(1,2)""".stripMargin,
+        sqlFilePath = "a.sql",
+        spark = spark
+      )
+    }
+
+    assert(ex.getMessage.contains("Specifying location is not supported"))
+  }
+
+  test("Tables living in arbitrary schemas can be read from pipeline") {
+    val database_name = "db_c215a150_c9c1_4c65_bc02_f7d50dea2f5d"
+    val table_name = s"$database_name.tbl"
+    spark.sql(s"CREATE DATABASE $database_name")
+    spark.sql(s"CREATE TABLE $table_name AS SELECT * FROM range(1,4)")
+
+    try {
+      val unresolvedDataflowGraph = unresolvedDataflowGraphFromSql(
+        sqlText = s"""
+                     |CREATE MATERIALIZED VIEW a AS SELECT * FROM $table_name;
+                     |CREATE STREAMING TABLE b AS SELECT * FROM STREAM($table_name);
+                     |CREATE TEMPORARY VIEW c AS SELECT * FROM a;
+                     |""".stripMargin
+      )
+
+      startPipelineAndWaitForCompletion(unresolvedDataflowGraph)
+
+      Seq("a", "b").foreach { tableName =>
+        assert(
+          spark
+            .sql(
+              s"SELECT * FROM ${fullyQualifiedIdentifier(tableName).quotedString}"
+            )
+            .collect()
+            .toSet == Set(1L, 2L, 3L).map(Row(_))
+        )
+      }
+    } finally {
+      spark.sql(s"DROP TABLE $table_name")
+      spark.sql(s"DROP DATABASE $database_name")
+    }
+  }
+
+  gridTest(s"Pipeline dataset can read from file based data sources")(
+    Seq("parquet", "orc", "json", "csv")
+  ) { fileFormat =>
+    val tmpDir = Utils.createTempDir().getAbsolutePath
+    spark.sql("SELECT * FROM RANGE(3)").write.format(fileFormat).mode("overwrite").save(tmpDir)
+
+    val unresolvedDataflowGraph = unresolvedDataflowGraphFromSql(
+      sqlText = s"""
+                   |CREATE MATERIALIZED VIEW a AS SELECT * FROM $fileFormat.`$tmpDir`;
+                   |CREATE STREAMING TABLE b AS SELECT * FROM STREAM($fileFormat.`$tmpDir`)
+                   |""".stripMargin
+    )
+
+    startPipelineAndWaitForCompletion(unresolvedDataflowGraph)
+
+    Seq("a", "b").foreach { datasetName =>
+      val datasetFullyQualifiedName =
+        fullyQualifiedIdentifier(datasetName).quotedString
+      spark.sql(s"REFRESH TABLE $datasetFullyQualifiedName")
+      val expectedRows = if (fileFormat == "csv") {
+        // CSV values are read as strings
+        Set("0", "1", "2")
+      } else {
+        Set(0, 1, 2)
+      }
+      assert(
+        spark.sql(s"SELECT * FROM $datasetFullyQualifiedName").collect().toSet == expectedRows.map(
+          Row(_)
+        )
+      )
+    }
+  }
+
+  test("Read from invalid data source would be categorized as reading an external table") {
```

Comment:
The test case name is confusing. Does it mean that file path table is not supported?